### PR TITLE
ci: :sparkles: Release to npm without third party action

### PR DESCRIPTION
**Pull request by lakshmaji-till**
## Description

Need to create a `tag` after change in `package.json` version.
- [x] Release package only on `master` branch with a commit pattern starting with `Release`, so both `github` and `npm` release will happens when commit is published. 
- [x] Removed `pascalgn/npm-publish-action@06e0830ea83eea10ed4a62654eeaedafb8bf50fc` which has some permissions issue https://github.com/hitz-group/dynamo-helper/actions/runs/4071706471/jobs/7013750208 

## Related Issues

Issue: #XXX

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by shidil — APPROVED** [review]

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -1,30 +1,26 @@
 name: npm-publish
 on:
   push:
-    branches:
-      - master
+    tags:
```
Comment (sravan1000) [inline .github/workflows/npm-publish.yml:4]: should we have any branch check here?

will it release without merging to master?

@shidil need your input here

Comment (shidil) [reply .github/workflows/npm-publish.yml:4]: Only when tags are created. It cannot be created by anyone, so fine

Comment (shidil) [reply .github/workflows/npm-publish.yml:4]: > will it release without merging to master?

That is possible

**Review by snehithv — DISMISSED** [review]

File: .github/workflows/npm-publish.yml (hunk 1 above)
Comment (lakshmaji-till) [reply .github/workflows/npm-publish.yml]: Added a few conditions which will make actions to be run when changes are merged to master and with a commit message `release`.
I still doubt `squash` will make this release disappear.

Comment (shidil) [reply .github/workflows/npm-publish.yml]: Tbh, I like the tag release better. This is fine as well

**Review by shidil — DISMISSED** [review]

**Review by sravan1000 — APPROVED** [review]

**Comment by lakshmaji-till** [comment]
Unit tests are failing due to memory usage, can any one help here!

**Review by snehithv — APPROVED** [review]

**Review by shidil — APPROVED** [review]

**Comment by shidil** [comment]
> Unit tests are failing due to memory usage, can any one help here!

I can take a look at it tomorrow.

**Pull request merged**